static/js/stream_ui_updates.js

```diff
@@ -162,6 +163,32 @@ export function update_permissions_banner(sub) {
     $settings.find(".stream-settings-tip-container").html(rendered_tip);
 }
 
+export function update_notifications_stream_in_settings() {
+    // if stream creation form is not open, return
+    if (!hash_util.is_create_new_stream_narrow()) {
+        return;
+    }
+
+    const notifications_stream = stream_data.get_notifications_stream();
+    const stream_privacy_symbol_html = render_stream_privacy({
+        invite_only: notifications_stream.invite_only,
+        is_web_public: notifications_stream.is_web_public,
+    });
+    const $replacement_html = $(
+        "<strong> " + stream_privacy_symbol_html + notifications_stream + "</strong>",
```

Comment:
I think it would be better to instead re-render the `announce_stream` template itself as it would be much cleaner way of doing this.